crates/build/src/lint.rs

```diff
@@ -36,7 +36,7 @@ pub const TOOLCHAIN_VERSION: &str = "nightly-2025-02-20";
 /// Git repository with ink_linting libraries
 pub const GIT_URL: &str = "https://github.com/use-ink/ink";
 /// Git revision number of the linting crate
-pub const GIT_REV: &str = "87a97b244f7eb30fe04b9dba59294af9f91646d4";
+pub const GIT_REV: &str = "5dc1624c3fa279cc57b4418b939cc089790f42f0";
```

Comment:
The actual fix required to resolve the build issue. A lot of manual debugging to figure this out. Should perhaps be supported by a test, but imagine this might change to crates.io releases so perhaps not worth the effort.